### ISSUE TITLE
azure-monitor-opentelemetry-exporter not compatible with latest opentelemetry-[api/sdk] 1.15.0

### ISSUE THREAD

**Issue by jabbera** [Azure/azure-sdk-for-python#27900]
- **Package Name**:  azure-monitor-opentelemetry-exporter
- **Package Version**:  1.0.0b10
- **Operating System**: Linux
- **Python Version**:  3.10

**Describe the bug**
not compatible with opentelemetry-[api/sdk] 1.15.0

**To Reproduce**
1. pip install azure-monitor-opentelemetry-exporter
2. python3.10 
3. from azure.monitor.opentelemetry.exporter import AzureMonitorLogExporter, AzureMonitorTraceExporter

see:
```
.tox/py310-sqlalchemy14-integration/lib/python3.10/site-packages/gmo/core/util/telemetry/__init__.py:8: in <module>
    from azure.monitor.opentelemetry.exporter import AzureMonitorLogExporter, AzureMonitorTraceExporter
.tox/py310-sqlalchemy14-integration/lib/python3.10/site-packages/azure/monitor/opentelemetry/exporter/__init__.py:7: in <module>
    from azure.monitor.opentelemetry.exporter.export.logs._exporter import AzureMonitorLogExporter
.tox/py310-sqlalchemy14-integration/lib/python3.10/site-packages/azure/monitor/opentelemetry/exporter/export/logs/_exporter.py:8: in <module>
    from opentelemetry.sdk._logs.severity import SeverityNumber
E   ModuleNotFoundError: No module named 'opentelemetry.sdk._logs.severity'
```

**Expected behavior**
import to work

**Screenshots**

**Additional context**

**Comment by kashifkhan** [comment]
Thank you for your feedback @jabbera . We will investigate asap and get back to you

**Comment by jeremydvoss** [comment]
The PR is merged. However, since this is an ongoing issue, let's keep this open until the next release. I am looking into whether we can do a release before January. In the meantime, use OTel 1.14

**Comment by jeremydvoss** [comment]
There's another issue: when using the fixed version of the exporter with OTel 1.15, I consistently get the following warning:
...\site-packages\werkzeug\serving.py:716: ResourceWarning: unclosed <socket.socket fd=1304, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0> self.socket = socket.fromfd(fd, address_family, socket.SOCK_STREAM) ResourceWarning: Enable tracemalloc to get the object allocation traceback

I've determined that no commit in exporter 1.0.0b11 has caused this. It's too early to know. But it seems to be an issue with how opentelemetry-instrumentation-flask==0.36b0 or opentelemetry-instrumentation-wsgi==0.36b0 use Werkzeug. I've confirmed the fixed version of the exporter does still send telemetry correctly. However, since the new version needs to be pinned to 1.15, we'll be encouraging people to use 1.15 seems to have some issue that needs to be addressed.

I am not able to release the exporter as is because of the memory issue with OTel 1.15. Instead, we can pin the exporter to 1.12<=x<=1.14 before the module path was changed. That way, we can avoid the memory allocation issue as well as the severity import breaking change.
